config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/string/CharacterBlacklistUtil.java

```diff
@@ -73,5 +78,58 @@ public static boolean containsInvalidCharacter(String name) {
         return false;
     }
 
+    /**
+     * Returns URL encoded name (even asterisk is encoded)
+     */
+    public static String encodeName(String name) throws UnsupportedEncodingException {
+        String encodedName = URLEncoder.encode(name, "UTF-8");
+        return encodedName.replaceAll("\\*", "%2A");
+    }
+
+    /**
+     * Returns URL encoded path (even asterisk is encoded)
+     */
+    public static String encodePath(String path) throws UnsupportedEncodingException {
+        if (path.equals("/")) {
+            return path;
+        }
+        final String[] folderNames = path.split("/");
+        StringBuilder pathBuilder = new StringBuilder();
+        for (String folderName : folderNames) {
+            pathBuilder.append(encodeName(folderName));
+            pathBuilder.append("/");
+        }
+        if (!path.endsWith("/")) {
+            pathBuilder.deleteCharAt(pathBuilder.length()-1);
+        }
+        return pathBuilder.toString();
+    }
+
+    /**
+     * Returns URL decoded path
+     */
+    public static String decodePath(String path) throws UnsupportedEncodingException {
+        if (path.equals("/")) {
+            return path;
+        }
+        final String[] folderNames = path.split("/");
+        StringBuilder pathBuilder = new StringBuilder();
+        for (String folderName : folderNames) {
+            pathBuilder.append(decodeName(folderName));
+            pathBuilder.append("/");
+        }
+        if (!path.endsWith("/")) {
+            pathBuilder.deleteCharAt(pathBuilder.length()-1);
+        }
+        return pathBuilder.toString();
+    }
+
+    /**
+     * Returns URL decoded name
+     */
+    public static String decodeName(String name) throws UnsupportedEncodingException {
+        return URLDecoder.decode(name, "UTF-8");
```

Comment:
The chances of this throwing an error are very low so I think you should convert this checked exception with the RuntimeException or just catch and ignore it here itself as we are already ignoring it in caller methods.